lib/sockunion.c

```diff
@@ -588,23 +588,6 @@ static void __attribute__((unused)) sockunion_print(const union sockunion *su)
 	}
 }
 
-int in6addr_cmp(const struct in6_addr *addr1, const struct in6_addr *addr2)
```

Comment:
couldn't we just change the implementation of this so that it's not so ... awkward?
it seems nice to have the v6 address-specific signature, for instance.

Reply:
except of course we have IPV6_ADDR_CMP and a couple other places where we just use memcmp for v6 addresses in sockunion.c.  I've just converted them over to use that version.  All in all looks like a bunch of places could be fixed up.

Reply:
sure- it's not the use of memcmp() at the bottom level that I was asking about, it's about having a v6-specific signature that seems ... useful and helpful. Just a fan of apis that help folks use our data types in a natural sort of way.

Reply:
I don't see why IPV6_ADDR_CMP is not sufficient.  the custom function was incredibly slow, when I removed it in a 256 peer X 100k routes I dropped from ~3:30 convergence to 2:30 convergence.

Reply:
I understand that the byte-by-byte implementation was not good - fine, let's do it better.
the macro is just a macro, so no type-checking of its arguments - I like the idea of a strongly-typed utility (that has a good implementation). but it's ok - if you and Donatas are ok with the macro, I'm not lying on the tracks